ref-exchange/src/token_receiver.rs

```diff
@@ -34,16 +34,17 @@ impl Contract {
     ) -> Vec<(AccountId, Balance)> {
         // [AUDIT_12] always save back account for a resident user
         let mut is_resident_user: bool = true;
-        let mut initial_account: Account = self.accounts.get(sender_id).unwrap_or_else(|| {
+        let initial_account: Account = self.accounts.get(sender_id).unwrap_or_else(|| {
             is_resident_user = false;
             if !force {
                 env::panic(ERR10_ACC_NOT_REGISTERED.as_bytes());
             } else {
                 Account::default().into()
             }
         }).into();
-        initial_account.deposit(&token_in, amount_in);
         let mut account = initial_account.clone();
+        // initial_account.deposit(&token_in, amount_in);
+        account.deposit(&token_in, amount_in);
```

Comment:
That's the interesting part. True it is nonsense if user deposit a token but start swap-action with another token. But it won't break the logic, right? if he deposit nETH and start a swap of BANANA to nUSDT, as the current code logic, when instant-swap complete (if he has banana in his inner account), all nETH amount he deposited would refund to him. (cause we take the  snapshot before counting in his deposit.)

Reply:
Now with a pure sandbox, instant swap user can't interfere inner accounts. So, if the first action is not start with the deposit token, it would simply fail with `token not registered` in sandbox.